patterns-use-cases/payment-state-machine/payment-state-machine-java/src/main/java/my/example/utils/TypeChecks.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
+ *
+ * This file is part of the Restate examples,
+ * which is released under the MIT license.
+ *
+ * You can find a copy of the license in the file LICENSE
+ * in the root directory of this repository or package or at
+ * https://github.com/restatedev/examples/
+ */
+package my.example.utils;
+
+import dev.restate.sdk.common.TerminalException;
+import my.example.types.Payment;
+
+public class TypeChecks {
+
+    public static void validatePayment(Payment payment){
+        if(payment.getAccountId() == null || payment.getAccountId().isEmpty()){
+            throw new TerminalException("Account ID is required");
+        }
+        if(payment.getAmountCents() <= 0){
+            throw new TerminalException("Amount must be greater than 0");
+        }
```

Comment:
You could have those checks in the constructor of the `Payment` directly. It's a bit more cleaner and you don't need another class.